rclcpp/CMakeLists.txt

```diff
@@ -212,442 +212,10 @@ ament_export_dependencies(statistics_msgs)
 ament_export_dependencies(tracetools)
 
 if(BUILD_TESTING)
-  find_package(ament_cmake_gtest REQUIRED)
   find_package(ament_lint_auto REQUIRED)
   ament_lint_auto_find_test_dependencies()
```

Comment:
why not putting this into the subfolder as well?

Reply:
It mirrors the organization of rcl's CMakeLists.txt. I can see it making sense either way. `ament_lint_auto` tests don't exactly belong to the tests performed in the test directory, so maybe they should be at the top level? On the other hand if they were in the test subdirectory, it just puts all tests in one location. 

https://github.com/ros2/rcl/blob/9e5bcd2965626b04e5ccc626d97b83b73231e913/rcl/CMakeLists.txt#L116-L121

Reply:
fair enough to mirror the style of RCL. I was mainly curious why not all test related cmake code would go into its own file. I guess semantically, one could argue that linters count as tests as well and could thus land in the subfolder. But I guess that's bike shedding here.